cogs/birthday.py

```diff
@@ -0,0 +1,97 @@
+            if len(querry) is 0:  # No Querry
+                results = db.execute("SELECT userId, date, month FROM birthdays").fetchall()
+                text = ""
+                for result in results:
+                    text += "User: {}\t->\t{}.{}.\n".format(result[0], result[1], result[2])
+                await ctx.send(text)
+            elif self.DATEPATTERN.match(querry) is not None:  # Birthday as Querry
+                day, month = querry.strip(".").split(".")
+                results = db.execute(
+                    "SELECT userId, date, month FROM birthdays WHERE date = {} AND month = {}".format(day,
+                                                                                                      month)).fetchall()
+                if len(results) is 0:
+                    await ctx.send("No Birthday at {}.{}.".format(day, month))
+                    return
+                text = ""
+                for result in results:
+                    text += "User: {}\t->\t{}.{}.\n".format(result[0], result[1], result[2])
+                await ctx.send(text)
+            else:  # Username as Querry
+                results = db.execute(
+                    "SELECT userId, date, month FROM birthdays WHERE userId LIKE '{}'".format(querry)).fetchall()
+                print(results)
+                if len(results) is 0:
+                    await ctx.send("No UserID matches this request")
+                    return
+                await ctx.send("User: {}\t->\t{}.{}.\n".format(results[0][0], results[0][1], results[0][2]))
+
+    @birthdays.command()
+    async def add(self, ctx, birthdate):
+        """adds a new bithday <DD.MM.> to the database, if possible"""
+        if self.DATEPATTERN.match(birthdate) is None:
+            await ctx.send("Usage: <!birthday add DD.MM.> (of course with a valid date)")
+            return
+        day, month = birthdate.strip(".").split(".")
+        self.update_birthday(ctx.message.author.id, day, month)
+        await ctx.message.add_reaction('\U00002705')
+
+    def get_current_date(self) -> [int, int]:
+        date = datetime.datetime.now()
+        return date.day, date.month
+
+    def update_birthday(self, user_id, day, month):
+        with self.bot.db.get(self.BIRTHDAY_CHANNEL_ID) as db:
+            db.execute("INSERT OR REPLACE INTO birthdays (userId, date, month) VALUES ('{}', {}, {})".format(
+                user_id, day, month))
+
+    async def congratulate(self):
+        day, month = self.get_current_date()
+        text = "Geburtstage am {}.{}.:".format(day, month)
+        users = self.get_user_ids(day, month)
+        if len(users) == 0:
+            return
+        for user in self.get_user_ids(day, month):
+            text += ":tada: :fireworks: :partying_face: **Alles Gute zum Geburtstag**, <@!{}>" \
+                    " :partying_face: :fireworks: :tada:\n".format(user)
+        await self.bot.fetch_channel(self.BIRTHDAY_CHANNEL_ID).send(text)
```

Comment:
Let it post wherever the birthday date was originally set up (i.e. save the channel ID in the database)

Reply:
but that would mean, that birthday messages would fly left right and center (metaphorically speaking), 
instead of in just one specific channel, that can be easily silenced

Reply:
i added a setup command to move this module to another channel, but I have no idea how to solve this properly..

Reason:
-  this Variable should be fixed -> 
-  must be determined at startup.->
-  a sql querry would be needed.. but since I cant have one static database (to store which channel to write to...)